### ISSUE TITLE
Benchmark application with `use_xnnpack=true`: ModifyGraphWithDelegate is disallowed when graph is immutable.

### ISSUE THREAD

**Issue by raryanpur** [tensorflow/tensorflow#41977]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux, Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Source
- TensorFlow version (use command below): v2.3.0
- Python version: N/A
- Bazel version (if compiling from source): 3.1.0
- GCC/Compiler version (if compiling from source): 7.5.0
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**

I've built the benchmarking application with XNNPack enabled for my host machine (Linux, Ubuntu 18.04) using the `tensorflow/tensorflow:2.3.0` Docker image. I built the application with the following bazel command

`bazel build -c opt --define tflite_with_xnnpack=true tensorflow/lite/tools/benchmark:benchmark_model`

When running `benchmark_application` with a non-quantized test model I have, TFLite spits out the following error: 

`ERROR: ModifyGraphWithDelegate is disallowed when graph is immutable.`

The benchmarking application still runs to completion, although I am not sure whether the XNNPack delegate is being used because of the above error. I noticed no difference in performance when running without `--use_xnnpack`, so don't think the delegate is being used.

I've run `benchmark_application` with the following command:

`./benchmark_model --graph=model.tflite --use_xnnpack=true`

**Describe the expected behavior**

Would expect with v2.3.0 for the XNNPack delegate to work with non-quantized tflite models. The model was generated with the TFLite model conversion tools, specifically the Keras to TFLite tools.

**Standalone code to reproduce the issue**

This may not be able to be reproduced across all platforms, however, invoking `benchmark_model` as above would be the best attempt to reproduce the issue.

The model can be found here: https://www.dropbox.com/s/dyed8ch0erghb1v/yolov4-416.tflite?dl=0

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

`uname -a` output: Linux ubuntu 4.15.0-36-generic #39~16.04.1-Ubuntu SMP Tue Sep 25 08:59:23 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Comment by multiverse-tf** [comment]
Note "bazel build -c opt --define tflite_with_xnnpack=true tensorflow/lite/tools/benchmark:benchmark_model" will compile the binary that will apply xnnpack delegate by default. 

If you omit "--define tflite_with_xnnpack=true" when building the benchmark tool, you could simply use "--use_xnnpack=true" to check the xnnpack delegate performance as you're aware of.

Speaking of the error, I think it's because we tried to apply xnnpack delegate twice in this case (i.e. the fist is triggered by setting "--use_xnnpack=true", and the second is triggered by applying xnnpack delegate by default (i.e. setting "--define tflite_with_xnnpack=true" when compiling the benchmark tool).

To tell if the xnnpack delegate has been applied, I think there should be sth. from the output that tells this.

Anyway, will look into this issue further by running the attached model.

**Comment by raryanpur** [comment]
@multiverse-tf thanks for following up.

> Note "bazel build -c opt --define tflite_with_xnnpack=true tensorflow/lite/tools/benchmark:benchmark_model" will compile the binary that will apply xnnpack delegate by default.

If I run `./benchmark_model --graph=model.tflite`, `Use xnnpack : [0]` is printed to stdout as part of `benchmark_model`'s debug output. Since the benchmarking tool was built with `--define tflite_with_xnnpack=true`, shouldn't I expect `Use xnnpack: [1]` to be output instead? If I run `./benchmark_model --graph=model.tflite --use_xnnpack=true`, `Use xnnpack: [1]` is output to stdout, in addition to (but not limited to) the following

`INFO: Created TensorFlow Lite XNNPACK delegate for CPU.`
`Applied XNNPACK delegate, and the model graph will be partially executed w/ the delegate.`

**Comment by multiverse-tf** [comment]
Hi Rameen,

Sorry for the confusing output message. "Use xnnpack: [..]" is an output from the model benchmark tool itself. Currently, there's no runtime query api into TFLite interpreter w.r.t. which delegates have been applied. As a result, when the xnnpack delegate could be applied by default (i.e. via compilation flag --define=tflite_with_xnnpack=true"), the benchmark tool doesn't know whether the xnnpack delegate has been applied or not by default by the underlying tflite interpreter.

We will provide more output messages from the tool, and hope it will further clarify the behavior.

**Comment by raryanpur** [comment]
@multiverse-tf thanks!

**Comment by jdduke** [comment]
Note: Let's leave this issue open to address the case where it should be perfectly OK to apply XNNPACK multiple times to the same graph (or to apply the GPU delegate, then the XNNPACK delegate).

**Comment by sushreebarsa** [comment]
@raryanpur It looks like you are using an older Version of Tensorflow . Many bugs have been fixed in the latest version. Could you please execute your code using Latest Version 2.4.1 or 2.5 and let us know if the issue still persists? Thanks!

**Comment by raryanpur** [comment]
@sushreebarsa @multiverse-tf provided an explantation of this behavior above that was fine for my use case at the time that I filed the bug and I should've closed the ticket then. It appears via @jdduke this ticket is relevant to the broader project so is being kept open. I don't currently have the bandwidth to test this for you on the latest version of TF but hopefully the details I've provided in the ticket description are enough for you or the team to test if desired.
